consensus/src/dag/reconstruction/mod.rs

```diff
@@ -72,8 +82,31 @@ impl<U: Unit> WrappedUnit<U::Hasher> for ReconstructedUnit<U> {
 }
 
 impl<U: Unit> UnitWithParents for ReconstructedUnit<U> {
-    fn parents(&self) -> &NodeMap<HashFor<Self>> {
-        &self.parents
+    fn parents(&self) -> impl Iterator<Item = &HashFor<U>> {
+        self.parents.values().map(|(hash, _)| hash)
+    }
+
+    fn direct_parents(&self) -> impl Iterator<Item = &HashFor<Self>> {
+        self.parents
+            .values()
+            .filter_map(|(hash, parent_round)| match self.unit.coord().round() {
+                0 => None,
```

Comment:
Maybe a short comment here for clarity that this never matters, since round 0 units cannot have nonempty parents.